src/Engine/Scene/SkeletonBasedAnimationSystem.cpp

```diff
@@ -0,0 +1,149 @@
+            auto endTaskId  = taskQueue->registerTask( endTask );
+            taskQueue->addPendingDependency( "AnimatorTask_" + skinComp->getSkeletonName(),
+                                             skinTaskId );
+            taskQueue->addDependency( skinTaskId, endTaskId );
+        }
+    }
+
+    m_time = frameInfo.m_animationTime;
+}
+
+void SkeletonBasedAnimationSystem::handleAssetLoading( Entity* entity,
+                                                       const Core::Asset::FileData* fileData ) {
+    auto skelData = fileData->getHandleData();
+    auto animData = fileData->getAnimationData();
+
+    // deal with AnimationComponents
+    Scalar startTime = std::numeric_limits<Scalar>::max();
+    Scalar endTime   = 0;
+    for ( const auto& skel : skelData )
+    {
+        auto component = new SkeletonComponent( "AC_" + skel->getName(), entity );
+        component->handleSkeletonLoading( skel );
+        component->handleAnimationLoading( animData );
+        auto [s, e] = component->getAnimationTimeInterval();
+        startTime   = std::min( startTime, s );
+        endTime     = std::max( endTime, e );
+        component->setXray( m_xrayOn );
+        registerComponent( entity, component );
+    }
+    // configure the time on the Engine
+    auto engine = RadiumEngine::getInstance();
+    engine->setStartTime( startTime );
+    engine->setEndTime( endTime );
+
+    // deal with SkinningComponents
+    auto geomData = fileData->getGeometryData();
+    if ( geomData.size() > 0 && skelData.size() > 0 )
+    {
+        for ( const auto& geom : geomData )
+        {
+            // look for a skeleton skinning this mesh
+            // warning: there should be at most one such skeleton!
+            auto it = std::find_if( skelData.begin(), skelData.end(), [&geom]( const auto& skel ) {
+                return std::find_if( skel->getBindMeshes().begin(),
+                                     skel->getBindMeshes().end(),
+                                     [&geom]( const auto& meshName ) {
+                                         return meshName == geom->getName();
+                                     } ) != skel->getBindMeshes().end();
+            } );
+            if ( it != skelData.end() )
+            {
+                const auto& skel             = *it;
+                SkinningComponent* component = new SkinningComponent(
+                    "SkC_" + geom->getName(), SkinningComponent::LBS, entity );
+                component->handleSkinDataLoading( skel, geom->getName() );
+                registerComponent( entity, component );
+            }
+        }
+    }
+}
+
+// Skeleton display
+
+void SkeletonBasedAnimationSystem::setXray( bool on ) {
+    m_xrayOn = on;
+    for ( const auto& comp : m_components )
+    {
+        if ( comp.second->getName().compare( 0, 3, "AC_" ) == 0 )
+        { static_cast<SkeletonComponent*>( comp.second )->setXray( on ); }
```

Comment:
b.t.w. here again we distinguish the `Component` kind through its name.

Reply:
oups ... will be corrected in the next push, and might be removed in a near future when XRay state willl be handle differently.